comp/netflow/server/component.go

```diff
@@ -0,0 +1,29 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+// Package server implements a component that runs the netflow server.
+// When running, it listens for network traffic according to configured
+// listeners and aggregates traffic data to send to the backend.
+// It does not expose any public methods.
+package server
+
+import (
+	"go.uber.org/fx"
+
+	"github.com/DataDog/datadog-agent/pkg/util/fxutil"
+)
+
+// team: network-device-monitoring
+
+// Component is the component type.
+type Component interface {
+	Start() error
```

Comment:
Can you use [lifecycle hooks](https://uber-go.github.io/fx/lifecycle.html#lifecycle-hooks) instead of having to explicitly call `Start` and `Stop`?

Reply:
I think I can do that. I had some issues in the tests where the race detector complained if e.g. I called EXPECT methods on a testing mock after the app was started, but I think I should be able to work around those by putting the expectations in an Invoke block instead of directly in the test.

Reply:
Done.